saltstack/base/salt/luks/init.sls

```diff
@@ -0,0 +1,98 @@
+/etc/luks:
+  file.directory:
+    - name: /etc/luks
+    - user: root
+    - group: root
+    - mode: 700
+
+/etc/luks/bin:
+  file.directory:
+    - name: /etc/luks/bin
+    - user: root
+    - group: root
+    - mode: 700
+
+/etc/luks/backup:
+  file.directory:
+    - name: /etc/luks/backup
+    - user: root
+    - group: root
+    - mode: 700
+
+luks_backing_file:
+  file.managed:
+    - name: /etc/luks/luks
+    - user: root
+    - group: root
+    - mode: 600
+    - create: true
+
+passphrase_ciphertext_file:
+  file.managed:
+    - name: /etc/luks/passphrase_ciphertext
+    - user: root
+    - group: root
+    - mode: 600
+    - create: true
+
+passphrase_encryption_key_file:
+  file.managed:
+    - name: /etc/luks/passphrase_encryption_key
+    - user: root
+    - group: root
+    - mode: 600
+    - create: true
+
+luks_mount_point:
+  file.directory:
+    - name: /mnt/luks
+    - user: root
+    - group: root
+    - mode: 700
+
+luks_passphrase_tmpfs_mount_point:
+  file.directory:
+    - name: /mnt/luks_passphrase_tmpfs
+    - user: root
+    - group: root
+    - mode: 700
+
+aws_encryption_sdk_logs:
+  file.directory:
+    - name: /var/log/luks
+    - user: root
+    - group: root
+    - mode: 700
+
+luks_create_script:
+  file.managed:
+    - name: /etc/luks/bin/create-luks-volume.sh
+    - source: salt://{{ slspath }}/bin/create-luks-volume.sh
+    - user: root
+    - group: root
+    - mode: 700
+
+luks_reopen_script:
+  file.managed:
+    - name: /etc/luks/bin/reopen-luks-volume.sh
+    - source: salt://{{ slspath }}/bin/reopen-luks-volume.sh
+    - user: root
+    - group: root
+    - mode: 700
+
+luks_reopen_service:
+  file.managed:
+    - name: /etc/systemd/system/reopen-luks-volume.service
+    - user: root
+    - group: root
+    - makedirs: True
+    - source: salt://{{ slspath }}/etc/systemd/system/reopen-luks-volume.service
```

Comment:
What is the file mode by default? I guess 644 is fine if that is the default one.

Reply:
The default is defined by the umask of the salt process, which for our case looks to `0022`, so the default should then be `755` then. In that case setting the permissions to something like `644` might make sense here.